scripts/lint_and_format.py

```diff
@@ -0,0 +1,96 @@
+import os
+import sys
+import subprocess
+import argparse
+
+parser = argparse.ArgumentParser(
+    description="Format C/C++ code using clang-format."
```

Comment:
Ideally we should also add `ruff` as a linter for all of our Python scripts. It will be a good (additional) PR